middleware/http/include/http/outbound/state.h

```diff
@@ -221,6 +225,15 @@ namespace casual
 
          common::domain::Identity identity{ "http"};
 
+         inline void maybe_metric()
+         {
+            if( pending.requests.empty() || metric)
+            {
+               common::communication::device::blocking::send( common::communication::instance::outbound::service::manager::device(), metric.message());
+               metric.clear();
+            }
+         }
+
```

Comment:
`State` should never "do anything", other than mutate it's own state. 
This to reduce complexity. Other stuff actually do stuff and mutate state to reflect what's done.
Also, with this "rule" we don't need ask our self, where "to do this stuff" -> less confusion and more coherent code.
Move this to _handle_.

There might be stuff that is "done" in a few places in other `State` -> if so, it should be removed.